pkg/addons/default/addons.go

```diff
@@ -1,27 +1,41 @@
 package defaultaddons
 
 import (
+	"github.com/aws/aws-sdk-go/service/eks/eksiface"
 	"github.com/weaveworks/eksctl/pkg/kubernetes"
 )
 
-func DoAddonsSupportMultiArch(clientSet kubernetes.Interface, rawClient kubernetes.RawClientInterface, controlPlaneVersion string, region string) (bool, error) {
-	kubeProxyUpToDate, err := IsKubeProxyUpToDate(clientSet, controlPlaneVersion)
+type AddonInput struct {
+	RawClient           kubernetes.RawClientInterface
+	EKSAPI              eksiface.EKSAPI
+	ControlPlaneVersion string
+	Region              string
+}
```

Comment:
kube proxy needs all 4
coredns needs 3
aws-node needs 2

either keep it as this or:
````

type Struct KubeProxyAddon {
	RawClient           kubernetes.RawClientInterface
	EKSAPI              eksiface.EKSAPI
	ControlPlaneVersion string
	Region              string}

type Struct KubeProxyAddon {
  	RawClient           kubernetes.RawClientInterface
	ControlPlaneVersion string
	Region              string
}


type Struct AWSNodeAddon {
  	RawClient           kubernetes.RawClientInterface
	Region              string
}


````

I don't mind :shrug:

Reply:
or maybe just a struct for kube-proxy/coredns. Idk so many different ideas, I give up thinking.